src/app/space/create/deployments/deployments-donut/deployments-donut-chart/deployments-donut-chart.component.ts

```diff
@@ -10,17 +19,15 @@ import * as d3 from 'd3';
   templateUrl: './deployments-donut-chart.component.html',
   styleUrls: ['./deployments-donut-chart.component.less']
 })
-export class DeploymentsDonutChartComponent implements DoCheck, OnInit, OnChanges, OnDestroy {
+export class DeploymentsDonutChartComponent implements AfterViewInit, OnChanges, OnDestroy, OnInit {
 
-  @Input() pods: any[];
+  @Input() pods: any;
```

Comment:
Could you define an interface for this type so we have static type checking available on it?